cmd/broadcaster-cli/app/keyset/topup/topup.go

```diff
@@ -30,19 +27,60 @@ var Cmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
+
 		logger := helper.GetLogger()
 
 		wocClient := woc_client.New(woc_client.WithAuth(wocApiKey), woc_client.WithLogger(logger))
-		fundingKeySet, _, err := helper.GetKeySetsKeyFile(keyFile)
-		if err != nil {
-			return fmt.Errorf("failed to get key sets: %v", err)
-		}
 
-		err = wocClient.TopUp(context.Background(), !isTestnet, fundingKeySet.Address(!isTestnet))
+		keysFlag := viper.GetString("keys")
+		selectedKeys := strings.Split(keysFlag, ",")
+
+		var keys map[string]string
+		err = viper.UnmarshalKey("privateKeys", &keys)
 		if err != nil {
 			return err
 		}
-		logger.Info("top up complete", slog.String("address", fundingKeySet.Address(!isTestnet)))
+
+		var keySets []*keyset.KeySet
+
+		if len(keys) > 0 {
```

Comment:
Could you move this logic into a separate helper function here `cmd/broadcaster-cli/helper/functions.go` so that it doesn't get repeated in each command?

Reply:
yes makes sense will move it